File: UI/static/js/user_stats.js
$(document).ready(function () {

    if (sessionStorage.getItem('current_user_id') === null){
        window.location.href = '/'
        alert('Korisnik nije ulogovan')
    }

    $.ajax({
        url: 'http://127.0.0.1:5001/load-profile',
        type: 'GET',
        data: {
            "id": sessionStorage.getItem('current_user_id')
        },
        success: function(response) {
            document.getElementById('userId').value = response.id;
            document.getElementById('userIdCard').value = response.id;
            document.getElementById('firstName').value = response.name;
            document.getElementById('lastName').value = response.lastname;
            document.getElementById('address').value = response.address;
            document.getElementById('city').value = response.city;
            document.getElementById('country').value = response.country;
            document.getElementById('phoneNum').value = response.phoneNumber;
            document.getElementById('email').value = response.email;
            document.getElementById('balance').value = response.balance;
            if(response.expDate != ""){
                document.getElementById('expDate').value = response.expDate;
            }
            if(response.nameOnCard != ""){
                document.getElementById('user').value = response.nameOnCard;
            }
            if(response.cardNumber != ""){
                document.getElementById('cardNumber').value = response.cardNumber;
            }
            if(response.verificated){
                document.getElementById('btnVerify').style.display = "none";
                document.getElementById('btnDeposit').style.display = "block";
            }
            else{
                document.getElementById('btnVerify').style.display = "block";
                document.getElementById('btnDeposit').style.display = "none";
            }
        }
    });

    $('#edit').submit(function(e){
        e.preventDefault();
        let id = $('#userId').val()
        let name = $('#firstName').val();
        let lastname = $('#lastName').val()
        let address = $('#address').val();
        let city = $('#city').val()
        let country = $('#country').val();
        let phoneNumber = $('#phoneNum').val()
        let email = $('#email').val();
        let password = $('#password').val()

        $.post('http://127.0.0.1:5001/update-profile', $('#edit').serialize(),
                        function (data, status) {
                            document.getElementById('userId').value = data.id;
                            document.getElementById('firstName').value = data.name
                            document.getElementById('lastName').value = data.lastname;
                            document.getElementById('address').value = data.address;
                            document.getElementById('city').value = data.city;
                            document.getElementById('country').value = data.country;
                            document.getElementById('phoneNum').value = data.phoneNumber;
                            document.getElementById('email').value = data.email;
                        });
    });

    $('#verification').submit(function(e){
        e.preventDefault();
        let id = $('#userId').val()
        let user = $('#user').val();
        let cardNumber = $('#cardNumber').val()
        let expDate = $('#expDate').val();
        let code = $('#code').val()
        
        $.post('http://127.0.0.1:5001/verify-account', $('#verification').serialize(),
                        function (data, status) {
                            alert("kartica uspesno dodata")
                            location.reload()
                        });
    });

    $('#btnDeposit').click(function () {
        money = $('#inputDeposit').val()
        if(money){
            $.post('http://127.0.0.1:5001/deposit-money', {'money': money, 'id': sessionStorage.getItem('current_user_id')},
            function (data, status) {
                document.getElementById('balance').value = JSON.stringify(data)
            });
        }

        document.getElementById('inputDeposit').value = 0
    });

});

File: Engine/blueprints/user_stats.py
from flask import Blueprint, render_template, request, flash, redirect, url_for, jsonify
from models.user import User
from db_config import db
import json
from werkzeug.security import generate_password_hash, check_password_hash

user_stats = Blueprint('user_stats', __name__)

@user_stats.route('/load-profile')
def load_profile():
    id = request.args.get('id')
    if id:
        #ako id postoji izvrsi upit nad bazom
        existing_user = User.query.get(int(id))
        return jsonify(existing_user.as_dict())
    return jsonify({})

@user_stats.route('/update-profile', methods=['POST'])
def update_profile():
    id = request.form.get('userId')
    name = request.form['firstName']
    lastname = request.form['lastName']
    email = request.form['email']
    password = request.form['password']
    address = request.form['address']
    city = request.form['city']
    country = request.form['country']
    phoneNumber = request.form['phoneNum']
    
    user = User.query.get(id)
    
    if password == "": #ako pass nije dodat onda uzimamo onaj koji se vec nalazi u korisniku
        update_user = User(id = id, name=name, lastname=lastname, email = email, password = user.password, address = address, city = city,
                       country = country, phoneNumber = phoneNumber, balance=user.balance, verificated = user.verificated, 
                       nameOnCard = user.nameOnCard, cardNumber = user.cardNumber, expDate = user.expDate)
    else:
        update_user = User(id = id, name=name, lastname=lastname, email = email, password = generate_password_hash(password), address = address, city = city,
                       country = country, phoneNumber = phoneNumber, balance=user.balance, verificated = user.verificated, 
                       nameOnCard = user.nameOnCard, cardNumber = user.cardNumber, expDate = user.expDate)
    
    db.session.delete(user)
    db.session.commit() #jer je id primarni kljuc a na isti id zelimo da upisemo i novog korisnika pa prvo moramo da
    #sacuvamo brisanje kako bi se id oslobodio da na njega upisemo izmene
    db.session.add(update_user)
    db.session.commit()
    return jsonify(update_user.as_dict())

@user_stats.route('/verify-account', methods=["POST"])
def verify():
    id = request.form.get('userIdCard')
    user = request.form.get('user')
    card_num = request.form.get('cardNumber')
    exp_date = request.form.get('expDate')
    code = request.form.get('code')
    
    user_by_id = User.query.get(id)
    user_by_id.verificated = True
    user_by_id.nameOnCard = user
    user_by_id.cardNumber = card_num
    user_by_id.expDate = exp_date
    db.session.commit()
    return jsonify(user_by_id.as_dict())
